src/rules/completedDocsRule.ts

```diff
@@ -19,100 +19,415 @@ import * as ts from "typescript";
 
 import * as Lint from "../index";
 
+export interface IBlockRequirementDescriptor {
+    visibilities?: Visibility[];
+}
+
+export interface IClassRequirementDescriptor {
+    locations?: Location[];
+    privacies?: Privacy[];
+}
+
+export type RequirementDescriptor = IBlockRequirementDescriptor | IClassRequirementDescriptor;
+
+export interface IRequirementDescriptors {
+    [type: string /* DocType */]: RequirementDescriptor;
+}
+
+export type All = typeof Rule.ALL;
+
+export type DocType = All
+    | typeof Rule.ARGUMENT_CLASSES
```

Comment:
What version of TypeScript are you on? For me (VS Code, TypeScript 2.1.X) `typeof Rule.ARGUMENT_CLASSES` correctly resolves to the `"classes"` literal.

Historically, before support for discriminated literal types, I liked having this explicit typing - both to inform the user and for future-proofing the type declarations.